### ISSUE TITLE
Integrate new parameters as az.rcParams

### ISSUE THREAD

**Issue by OriolAbril** [arviz-devs/arviz#792]
## Extend `az.rcParams`

Global options were recently added to ArviZ (see #734) following matplotlib's implementation. Currently though, only 3 parameters are integrated with ArviZ code. This issue can serve two purposes, first to list and keep track of possible rcParams ideas, and second point out that many of them are up for the grabs and to give some basic directions on implementation of new rcParams so that it becomes easier for the community to add new parameters. 

## How to extend rcParams

rcParams implementation is basically a python dictionary with 2 basic peculiarities. The first one is that keys are fixed; the dictionary is initialized when importing ArviZ and loads all parameters. Once this is done, new keys cannot be added and present keys cannot be deleted. The second one is that when trying to modify the value of one of the parameters, the new value is first validated before updating rcParams. This prevents errors in the ArviZ functions that use rcParams, as they know what to expect.

To implement new parameters, in addition to giving a look to the already implemented params, the following checklist should be followed:

* Create one validation function (or use one of the already present) 
* Add the parameter to the `defaultParams` dict.
* Add the parameter to the arvizrc.template (with a short description of what the param does)
* Modify the corresponding functions so that they use rcParams
* Document the new rcParam where appropriate (i.e. `data.load` is documented in `from_netcdf` methods)
* If a new validation function was created, add a test for it
* If possible, add a test for the new rcParam integration using `az.rc_context`

## Ideas

Here I leave an unfinished list with some ideas, please comment on possible additions to rcParams:

#### Data
* `data.stan_log_likelihood` allow to set a default name (like `log_lik`) for the log likelihood variable in Stan.

#### Stats
* ~`stats.credible_interval` Set the default credible interval, defaulting to 0.94.~
* `stats.pointwise_elpd` Default to false, but allow to set it to true in order to get pointwise elpd diagnostics by default (pointwise values are always calculated, so the only overhead would be in the creation of the `ELPDData` object)
* ~`stats.ic_scale`~
* `stats.compare_method` set the default for `method` argument in `az.compare`. Low priority, its main use would be backwards compatibility if we ever modified the default.

#### Plot
* ~`plot.point_estimate` set the default point estimate in `plot_density` and `plot_posterior`.~
* `plot.bandwidth` or `plot.bw` default bandwidth for kde

**Comment by ahartikainen** [comment]
`data.stan_log_likelihood` -> `data.elementwise_log_likelihood`/ `data.elem_log_likelihood`

Also other future methods with dictionary input could use it. Should the default be None

**Comment by OriolAbril** [comment]
Closing this, we can add new issues for any rcparam we might want to discuss
